tests/test_enhanced_transactions.rs

```diff
@@ -0,0 +1,262 @@
+    let helius = Helius {
+        config,
+        client,
+        rpc_client,
+    };
+    let request = ParseTransactionsRequest {
+        transactions: vec![
+            "DiG7v24AXRQqGagDx8pcVxRgVrgFoXUpJgp7xb62ycG9".to_string(),
+            "46tC8n6GyWvUjFxpTE9juG5WZ72RXADpPhY4S1d6wvTi".to_string(),
+        ],
+    };
+    server
+        .mock("POST", "/v0/transactions?api-key=fake_api_key")
+        .with_status(500)
+        .with_header("Content-Type", "application/json")
+        .with_body(r#"{"error":"Internal Server Error"}"#)
+        .create();
+
+    let response = helius.parse_transactions(request).await;
+    assert!(response.is_err(), "Expected an error due to server failure");
+}
+#[tokio::test]
+async fn test_parse_transaction_history_success() {
+    let mut server: Server = Server::new_with_opts_async(mockito::ServerOpts::default()).await;
+    let url: String = format!("{}/", server.url());
+
+    let mock_response = vec![EnhancedTransaction {
+        account_data: vec![AccountData {
+            account: "".to_string(),
+            native_token_balance: Some(Number::from(10)),
+            token_balance_changes: None,
+        }],
+        description: "Human readable interpretation of the transaction".to_string(),
+        transaction_type: TransactionType::Any,
+        source: Source::Jupiter,
+        fee: 5000,
+        fee_payer: "8cRrU1NzNpjL3k2BwjW3VixAcX6VFc29KHr4KZg8cs2Y".to_string(),
+        signature: "yy5BT9benHhx8fGCvhcAfTtLEHAtRJ3hRTzVL16bdrTCWm63t2vapfrZQZLJC3RcuagekaXjSs2zUGQvbcto8DK"
+            .to_string(),
+        slot: 148277128,
+        native_transfers: None,
+        token_transfers: Some(vec![TokenTransfer {
+            user_accounts: TransferUserAccounts {
+                from_user_account: Some("2iK5FbRZcJHAfhUNYmYdKTzSLnZE9NGAECurPoxDA3o7".to_string()),
+                to_user_account: Some("DCAKxn5PFNN1mBREPWGdk1RXg5aVH9rPErLfBFEi2Emb".to_string()),
+            },
+            from_token_account: Some("DCAKxn5PFNN1mBREPWGdk1RXg5aVH9rPErLfBFEi2Emb".to_string()),
+            to_token_account: Some("Bbnr95sKEcgWHHdD6UEU7MDek419KgMP1tWYUPP61fJk".to_string()),
+            token_amount: Number::from(32),
+            token_standard: TokenStandard::Fungible,
+            mint: "6oCioNHNTh4Xoz33mQSoTW4mxxmnyWVBNgv7zHjUuBkK".to_string(),
+        }]),
+        transaction_error: None,
+        instructions: vec![Instruction {
+            accounts: vec![],
+            data: "kdL8HQJrbbvQRGXmoadaja1Qvs".to_string(),
+            program_id: "MEisE1HzehtrDpAAT8PnLHjpSSkRYakotTuJRPjTpo8".to_string(),
+            inner_instructions: vec![InnerInstruction {
+                accounts: vec![],
+                data: "Dd1k91cWt84qJoQr3F".to_string(),
+                program_id: "Dd1k91cWt84qJoQr3FT7EXQpSaMtZtwPwdho7RbMWtEV".to_string(),
+            }],
+        }],
+        events: TransactionEvent::default(),
+        timestamp: 1656442333,
+    }];
+
+    let mock = server
```

Comment:
This is an unused variable. I think you might have meant to do `server.mock()` instead of declaring it as `mock`